packages/adminrouter/extra/src/includes/http/common.conf

```diff
@@ -1,9 +1,87 @@
-# Enable VTS module for metrics
-# Set the shared memory zone size to 32MB to be sure to
-# contain all metrics in the VTS module configuration.
-# If the VTS module configuration enables dynamic metric
-# keys this needs to be increased to accommodate for that.
-vhost_traffic_status_zone shared:vhost_traffic_status:32m;
+# Enable VTS module for metrics.
+#
+# The VTS module uses shared memory to store metrics data.
+# In particular, we store metrics data generated by the
+# ``vhost_traffic_status_filter_by_set_key`` directive.
+#
+# If this memory is used, Admin Router does not crash, but errors are shown
+# in the log, and metrics are no longer collected.
+# The amount of memory allocated here is allocated on each DC/OS node.
+# The number of metrics data items (called nodes) stored on each DC/OS node
+# scales with the number of DC/OS nodes in the cluster.
+#
+# Allocating more memory (and therefore more metrics nodes):
+# - (+) If we hit the maximum node count, we will remove old data - this only
+#       is an issue if this data has not been scraped, so the more memory we
+#       have, the more flexible we can be with the scraping interval.
+# - (-) Increased memory usage and requirements for DC/OS.
+# - (-) The more nodes there are, the slower the scrape takes, so the larger
+#       the scraping interval has to be.
+#
+# The shared memory zone size must be sufficiently large to ensure that all
+# metrics that could potentially be generated by the VTS module's
+# vhost_traffic_status_filter_by_set_key directive can be stored. If the
+# shared memory zone limit is hit Admin Router will not create any new metric
+# nodes and it will print error messages such as `ngx_slab_alloc() failed: no
+# memory in vhost_traffic_status_zone "ngx_http_vhost_traffic_status"`.
+#
+# See https://jira.mesosphere.com/browse/DCOS_OSS-4695 for an issue to create
+# a test which shows that Admin Router survives when the shared memory is full.
+#
+# Set shared memory size to 16MB. This accommodates ~4000 metric nodes. Reserve
+# 1300 metric nodes for Admin Router agent forwarding where each agent is
+# counted as distinct upstream in the order of at maximum 1000 agents.
+# Upstreams are seemingly not accounted for by the VTS module LRU mechanism so
+# we must always keep enough room for them.
+vhost_traffic_status_zone shared:vhost_traffic_status:16m;
+#
```

Comment:
It would be nice to have a blank line after each directive (and possibly before) to highlight the actual directives amongst the copious comments.